receiver/tlscheckreceiver/config.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package tlscheckreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/tlscheckreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/tlscheckreceiver/internal/metadata"
+)
+
+// Predefined error responses for configuration validation failures
+var (
+	errMissingURL = errors.New(`"url" must be specified`)
+	errInvalidURL = errors.New(`"url" must be in the form of <scheme>://<hostname>[:<port>]`)
+)
+
+// Config defines the configuration for the various elements of the receiver agent.
+type Config struct {
+	scraperhelper.ControllerConfig `mapstructure:",squash"`
+	metadata.MetricsBuilderConfig  `mapstructure:",squash"`
+	Targets                        []*targetConfig `mapstructure:"targets"`
+}
+
+type targetConfig struct {
+	URL string `mapstructure:"url"`
+}
+
+// Validate validates the configuration by checking for missing or invalid fields
+func (cfg *targetConfig) Validate() error {
+	var err error
+
+	if cfg.URL == "" {
+		err = multierr.Append(err, errMissingURL)
+	} else {
+		_, parseErr := url.ParseRequestURI(cfg.URL)
```

Comment:
What is the expected behavior (of validate && the receiver) for a non-https URL? I would expect it to be invalid at validate time, as I don't see a good mechanism for returning that information at runtime in a way that would be consumable.

Reply:
That is a good point and something definitely overlooked. 

For the sake of the PR being open so long and becoming harder to merge, I would love to see this as a follow up PR instead updating the current PR.

Reply:
I don't think we should allow for scheme. We do not need to send an http request here, only establish tcp connection, so scheme doesn't make sense. Related, I think we should change `url` to `host` since we are connecting to a host address via tcp, not sending a http request to a url.